os/fs/smartfs/smartfs_utils.c

```diff
@@ -300,6 +300,30 @@ void smartfs_wrle32(uint8_t *dest, uint32_t val)
 	smartfs_wrle16(dest + 2, (uint16_t)(val >> 16));
 }
 
+/****************************************************************************
+ * Name: is_path_last_slash
+ *
+ * Description:
+ *   Checks if the path for a directory entry ends with a '/' character
+ *
+ * Input Parameters:
+ *   path - the path to check for
+ *
+ * Returned Value:
+ *   boolean "true" if path ends with '/', otherwise "false"
+ *
+ ****************************************************************************/
+
+bool is_path_last_slash(const char* path)
+{
+	int i;
+	for (i = 0; path[i] != '\0'; i++);
```

Comment:
How about using `strlen`? It is almost same as libc `strlen`, but some arch specific code have much faster `strlen`.

Reply:
Updated